test/MonitorTest.hpp

```diff
@@ -65,20 +68,27 @@ class DirectoryMonitor {
 
         while (!exterminate) {
```

Comment:
spin/sleep locking is not the way to go. Don't use a thread inside the monitor. The monitor should be inside a thread in the GUI case, but not the other way around.